cli/pcluster/config/mapping.py

```diff
@@ -0,0 +1,530 @@
+    ec2_placement_group_validator,
+    ec2_security_group_validator,
+    ec2_subnet_id_validator,
+    ec2_volume_validator,
+    ec2_vpc_id_validator,
+    efa_validator,
+    efs_id_validator,
+    efs_validator,
+    fsx_id_validator,
+    fsx_imported_file_chunk_size_validator,
+    fsx_storage_capacity_validator,
+    fsx_validator,
+    raid_volume_iops_validator,
+    scheduler_validator,
+    url_validator,
+)
+
+
+class CommonRegex(Enum):
+    """Utility class containing all the common regex used in the section mapping."""
+
+    CIDR = r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/(0|1[6-9]|2[0-9]|3[0-2])$"
+    SECURITY_GROUP_ID = r"^sg-[0-9a-z]{8}$|^sg-[0-9a-z]{17}$"
+    SUBNET_ID = r"^subnet-[0-9a-z]{8}$|^subnet-[0-9a-z]{17}$"
+
+
+AWS = {
+    "type": Section,
+    "key": "aws",
+    "params": {
+        "aws_access_key_id": {},
+        "aws_secret_access_key": {},
+        "aws_region_name": {
+            "default": "us-east-1",  # TODO add regex
+        },
+    }
+}
+
+ALIASES = {
+    "type": Section,
+    "key": "aliases",
+    "params": {
+        "ssh": {
+            "default": "ssh {CFN_USER}@{MASTER_IP} {ARGS}"
+        },
+    }
+}
+
+SCALING = {
+    "type": Section,
+    "key": "scaling",
+    "label": "default",
+    "params": {
+        "scaledown_idletime": {
+            "type": IntParam,
+            "default": 10,
+            "cfn": "ScaleDownIdleTime",
+        }
+    }
+}
+
+VPC = {
+    "type": Section,
+    "key": "vpc",
+    "label": "default",
+    "params": {
+        "vpc_id": {
+            "cfn": "VPCId",
+            "allowed_values": r"^vpc-[0-9a-z]{8}$|^vpc-[0-9a-z]{17}$",
+            "validator": ec2_vpc_id_validator,
+        },
+        "master_subnet_id": {
+            "cfn": "MasterSubnetId",
+            "allowed_values": CommonRegex.SUBNET_ID.value,
+            "validator": ec2_subnet_id_validator,
+        },
+        "ssh_from": {
+            "default": "0.0.0.0/0",
+            "allowed_values": CommonRegex.CIDR.value,
+            "cfn": "AccessFrom",
+        },
+        "additional_sg": {
+            "cfn": "AdditionalSG",
+            "allowed_values": CommonRegex.SECURITY_GROUP_ID.value,
+            "validator": ec2_security_group_validator,
+        },
+        "compute_subnet_id": {
+            "cfn": "ComputeSubnetId",
+            "allowed_values": CommonRegex.SUBNET_ID.value,
+            "validator": ec2_subnet_id_validator,
+        },
+        "compute_subnet_cidr": {
+            "cfn": "ComputeSubnetCidr",
+            "allowed_values": CommonRegex.CIDR.value,
+        },
+        "use_public_ips": {
+            "type": BoolParam,
+            "default": True,
+            "cfn": "UsePublicIps",
+        },
+        "vpc_security_group_id": {
+            "cfn": "VPCSecurityGroupId",
+            "allowed_values": CommonRegex.SECURITY_GROUP_ID.value,
+            "validator": ec2_security_group_validator,
+        },
+        "master_availability_zone": {
```

Comment:
The PclusterConfig object should be an object containing a generic cluster configuration not related to the fact we can convert it to CFN.

I created an internal configuration parameter for master_availability_zone because it is used for CFN creation but also to verify if a configured EFS has a mount target in the master avail zone.